Mathlib/Algebra/Group/Pointwise/Set/Card.lean

```diff
@@ -3,60 +3,89 @@ Copyright (c) 2024 Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yaël Dillies
 -/
-import Mathlib.Algebra.Group.Pointwise.Finset.Basic
+import Mathlib.Data.Set.Pointwise.Finite
 import Mathlib.SetTheory.Cardinal.Finite
 
 /-!
-# Cardinalities of pointwise operations on sets.
+# Cardinalities of pointwise operations on sets
 -/
 
-namespace Set
-
-open Pointwise
-
-variable {α β : Type*}
+open scoped Cardinal Pointwise
 
-section MulAction
-variable [Group α] [MulAction α β]
+namespace Set
+variable {G M α : Type*}
 
-@[to_additive (attr := simp)]
-lemma card_smul_set (a : α) (s : Set β) : Nat.card ↥(a • s) = Nat.card s :=
-  Nat.card_image_of_injective (MulAction.injective a) _
+section Mul
+variable [Mul M] {s t : Set M}
 
-end MulAction
+@[to_additive]
+lemma _root_.Cardinal.mk_mul_le : #(s * t) ≤ #s * #t := by
+  rw [← image2_mul]; exact Cardinal.mk_image2_le
 
-section IsCancelMul
-variable [Mul α] [IsCancelMul α] {s t : Set α}
+variable [IsCancelMul M]
 
 @[to_additive]
-lemma card_mul_le : Nat.card (s * t) ≤ Nat.card s * Nat.card t := by
+lemma natCard_mul_le : Nat.card (s * t) ≤ Nat.card s * Nat.card t := by
   classical
   obtain h | h := (s * t).infinite_or_finite
   · simp [Set.Infinite.card_eq_zero h]
   obtain ⟨hs, ht⟩ | rfl | rfl := finite_mul.1 h
-  · lift s to Finset α using hs
-    lift t to Finset α using ht
-    rw [← Finset.coe_mul]
-    simpa [-Finset.coe_mul] using Finset.card_mul_le
+  · unfold Nat.card
+    rw [← Cardinal.toNat_mul]
+    gcongr
+    · exact Cardinal.mul_lt_aleph0 hs.lt_aleph0 ht.lt_aleph0
+    · exact Cardinal.mk_mul_le
   all_goals simp
 
-end IsCancelMul
+@[to_additive (attr := deprecated (since := "2024-09-30"))] alias card_mul_le := Cardinal.mk_mul_le
+
+end Mul
 
 section InvolutiveInv
-variable [InvolutiveInv α] {s t : Set α}
+variable [InvolutiveInv G] {s t : Set G}
+
+@[to_additive (attr := simp)]
+lemma _root_.Cardinal.mk_inv (s : Set G) : #↥(s⁻¹) = #s := by
+  rw [← image_inv, Cardinal.mk_image_eq_of_injOn _ _ inv_injective.injOn]
 
 @[to_additive (attr := simp)]
-lemma card_inv (s : Set α) : Nat.card ↥(s⁻¹) = Nat.card s := by
+lemma natCard_inv (s : Set G) : Nat.card ↥(s⁻¹) = Nat.card s := by
   rw [← image_inv, Nat.card_image_of_injective inv_injective]
 
+@[to_additive (attr := deprecated (since := "2024-09-30"))] alias card_inv := natCard_inv
+
 end InvolutiveInv
 
+section DivisionMonoid
+variable [DivisionMonoid M] {s t : Set M}
+
+@[to_additive]
+lemma _root_.Cardinal.mk_div_le : #(s / t) ≤ #s * #t := by
+  rw [div_eq_mul_inv, ← Cardinal.mk_inv t]; exact Cardinal.mk_mul_le
```

Comment:
optional - is it not easier to prove this by going via image2?

Reply:
Well spotted